Update docs for process filter (#118)
mesg-foundation/docs: guide/process/steps/filter.md

- `.` to access data in an object.
- `[INDEX]` to access an element in an array at the index `INDEX`. (You might get a runtime error if your index is out of bounds).

#### Examples
  - `foo`: Access the variable `foo`.
  - `foo.bar`: Access the variable `bar` in the object `foo`.
  - `foo[0]`: Access the first element of the array `foo`.
  - `foo[0].bar[1]`: Access the second element of the array `bar` contained in the first element of the array `foo`.
  - `foo[0][1]`: Access the second element in the first element of the array `foo`.

```yaml
- type: filter
  conditions:
    - key: foo
      predicate: EQ
      value: 2
    - key: data.foo[0].x
      predicate: EQ
      value: 2
```

### Predicate

The following predicates are available:

- `EQ`: The value must be exactly equal to the data. Works with types `string`, `bool`, and `number`.
- `GT`: The value must be greater than the data. Works only with types `number`.
- `GTE`: The value must be greater than or equal to the data. Works only with types `number`.
- `LT`: The value must be lesser than the data. Works only with types `number`.
- `LTE`: The value must be lesser than or equal to the data. Works only with types `number`.

> - `CONTAINS`: The value must be contained in the data. Works with any type of value be only with **repeated** data.

Reviewer: ```suggestion
- `CONTAINS`: The value must be contained in the data. Works with `string` and **repeated** data.
```

Author: fixed
